### PR TITLE
MGMT-11309: clean up prom metrics to enable meaningful alerts

### PR DESCRIPTION
**Pull request by rccrdpccl**
This PR aims to reduce number of unnecessary labels for exported prometheus metrics.
The reason is that we want to be able to alert if our statistics (elasticsearch) falls behind (clusters, events, etc) compared to the service itself.
Once this gets merged, we will add a follow-up PR that will only add exporting event count, to enable comparison between the two systems with the most meaningful stat.

## List all the issues related to this PR

- [ ] New Feature 
- [X] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [X] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @gamli75 
/cc @osherdp 
/cc @jhernand 

## Checklist

- [X] Title and description added to both, commit and PR.
- [X] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [X] Reviewers have been listed
- [X] This change does not require a documentation update (docstring, `docs`, README, etc)
- [X] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/metrics/metricsManager.go [hunk 1]
```diff
@@ -45,18 +42,16 @@ const (
 )
 
 const (
-	counterDescriptionClusterCreation                        = "Number of cluster resources created, by version"
-	counterDescriptionClusterInstallationStarted             = "Number of clusters that entered installing state, by version"
-	counterDescriptionClusterHostInstallationCount           = "Number of hosts per cluster"
-	counterDescriptionClusterHostNTPFailuresCount            = "Number of NTP failures per cluster"
```
Comment (rccrdpccl) [inline internal/metrics/metricsManager.go:51]: This seems not to be exported in any case

File: internal/metrics/metricsManager_test.go [hunk 2]
```diff
@@ -14,8 +14,8 @@ import (
 )
 
 var _ = DescribeTable(
-	"Adds agent version label to host installation phase seconds metric",
-	func(agentVersion string) {
+	"Tests that label has been added as expected",
```
Comment (rccrdpccl) [inline internal/metrics/metricsManager_test.go:17]: This test is not extremely useful as it stands, but it is a starting point for testing metrics.
As the previous tests was not needed anymore, I think it's best to leave a simple test to make it easier for contributors to add new tests

Comment (jhernand) [reply internal/metrics/metricsManager_test.go:17]: I agree, looks good to me.

File: internal/metrics/metricsManager_test.go [hunk 3]
```diff
@@ -68,9 +67,9 @@ var _ = DescribeTable(
 		// Verify that the label was added:
 		metrics := server.Metrics()
 		Expect(metrics).To(MatchLine(
-			`^service_assisted_installer_host_installation_phase_seconds_count\{.*discoveryAgentVersion="v1.0.0-141".*\} .*$`,
+			expectedMetricRegex,
 		))
 	},
-	Entry("Full image reference", "quay.io/edge-infrastructure/assisted-installer:v1.0.0-141"),
-	Entry("Tag only", "v1.0.0-141"),
+	Entry("Starting installation phase", models.HostStageStartingInstallation, `^service_assisted_installer_host_installation_phase_seconds_count\{.*phase="Starting installation".*\} .*$`),
```
Comment (jhernand) [inline internal/metrics/metricsManager_test.go:73]: Not really relevant, but I believe that very long lines are hard to read and change. I'd suggest to put each argument in a separate line:

```go
Entry(
        "Starting installation phase",
        models.HostStageStartingInstallation,
        `...`,
),
```

**Comment by gamli75** [comment]
As part of this cleanup - we must clean up our grafana dashboards and make sure that we have the same dashboards in elastic, see:
https://grafana.app-sre.devshift.net/d/assisted-installer-overview/general-overview?orgId=1

**Comment by osherdp** [comment]
/uncc
seems to have enough reviewers

**Comment by gamli75** [comment]
Another remark - those counters will be reset every time we will deploy a new version or someone will restart the service

**Comment by rccrdpccl** [comment]
> Another remark - those counters will be reset every time we will deploy a new version or someone will restart the service
Yes, this is to be expected.
The idea is to leverage prometheus function [increase](https://prometheus.io/docs/prometheus/latest/querying/functions/#increase) to perform the checks

**Review by gamli75 — APPROVED** [review]

**Pull request merged**